gnomad_qc/v3/subset.py

```diff
@@ -0,0 +1,291 @@
+    hl.init(
+        log="/subset.log",
+        default_reference="GRCh38",
+        tmp_dir="gs://gnomad-tmp/subset_v3",
+    )
+    hl._set_flags(no_whole_stage_codegen="1")
+    hard_filter = args.hard_filter_samples
+    release_only = args.release_only
+    output_path = args.output_path
+    vds_output = args.vds
+    dense_mt = args.dense_mt
+    split_multi = args.split_multi
+    subset_call_stats = args.subset_call_stats
+    add_variant_qc = args.add_variant_qc
+    variant_qc_annotations = args.variant_qc_annotations
+    pass_only = args.pass_only
+    n_partitions = args.num_partitions
+    vcf = args.vcf
+    header_dict = HEADER_DICT
+
+    try:
+        if (add_variant_qc or pass_only) and not split_multi:
+            raise ValueError(
+                "Cannot annotate variant QC annotations or filter to PASS variants on"
+                " an unsplit dataset."
+            )
+
+        if n_partitions and vds_output:
+            raise ValueError(
+                "Number of partitions argument is not supported for VDS output at this"
+                " time."
+            )
+
+        vds = get_gnomad_v3_vds(
+            remove_hard_filtered_samples=hard_filter,
+            release_only=release_only,
+        )
+
+        if args.test:
+            vds = hl.vds.VariantDataset(
+                vds.reference_data._filter_partitions(range(2)),
+                vds.variant_data._filter_partitions(range(2)),
+            )
+
+        subset_ht = hl.import_table(args.subset_samples).key_by("s")
+        logger.info("Subsetting to %d samples.", subset_ht.count())
+        vds = hl.vds.filter_samples(vds, subset_ht, remove_dead_alleles=True)
+        logger.info(
+            "Final number of samples being kept in the VDS: %d.",
+            vds.variant_data.count_cols(),
+        )
+
+        if split_multi:
+            logger.info("Splitting multiallelics.")
+            vd = vds.variant_data
+            vd = vd.annotate_rows(
+                n_unsplit_alleles=hl.len(vd.alleles),
+                mixed_site=(hl.len(vd.alleles) > 2)
+                & hl.any(lambda a: hl.is_indel(vd.alleles[0], a), vd.alleles[1:])
+                & hl.any(lambda a: hl.is_snp(vd.alleles[0], a), vd.alleles[1:]),
+            )
+            vds = hl.vds.split_multi(
+                hl.vds.VariantDataset(vds.reference_data, vd), filter_changed_loci=True
+            )
+
+            if add_variant_qc or pass_only:
+                vd = vds.variant_data
+                ht = release_sites().ht()
+                if pass_only:
+                    logger.info("Filtering to variants that passed variant QC.")
+                    vd = vd.filter_rows(ht[vd.row_key].filters.length() == 0)
+                if add_variant_qc:
+                    logger.info("Adding variant QC annotations.")
+                    vd = vd.annotate_rows(
+                        **make_variant_qc_annotations_dict(
```

Comment:
This function from v4 is using v4 release_sites HT, but `raw_qual_hists` is now under `histograms`, maybe copy that function here, and use v3 release_sites HT?
It will run into error otherwise.